### PR TITLE
Removed intern dependencies from abstract to flat strings

### PR DESCRIPTION
**Pull request by R4PaSs**
As @privat suggested, here's the PR that cleans the abstract Text class of the string block, this will serve as prelude for the integration of #1400

### CODE REVIEW

File: lib/standard/string.nit [hunk 1]
```diff
@@ -1537,6 +1532,10 @@ end
 abstract class Buffer
 	super Text
 
+	new do return new FlatBuffer
```
Comment (privat) [inline lib/standard/string.nit]: doc?

File: lib/standard/string.nit [hunk 2]
```diff
@@ -1537,6 +1532,10 @@ end
...
+
+	new with_cap(i: Int) do return new FlatBuffer.with_capacity(i)
```
Comment (privat) [inline lib/standard/string.nit]: doc?... and with not `with_capacity`

**Comment by R4PaSs** [comment]
Reroll, added doc and fixed failing nitunit

**Comment by Morriar** [comment]
+1

**Comment by R4PaSs** [comment]
Reroll, tests should be fixed, I tried the curl example locally, there seems to be no problem with it, maybe a temporary downtime when jenkins ran the test

**Comment by R4PaSs** [comment]
test this please

**Comment by R4PaSs** [comment]
Reroll: Tests passing and fixed merge conflict due to #1385

**Comment by privat** [comment]
+1

**Comment by Morriar** [comment]
+1

**Pull request merged**